tests/test_flow.py

```diff
@@ -2556,6 +2556,19 @@ def test_custom_string_time_input():
     assert result.time == datetime(2021, 5, 9, 14, 5, 27, tzinfo=pytz.utc)
 
 
+def test_none_key_is_not_written():
+    data = pd.DataFrame({'first_name': ['moshe', None, 'katya'], 'some_data': [1, 2, 3]})
+    data.set_index(keys=['first_name'], inplace=True)
+
+    controller = build_flow([
+        DataframeSource(data, key_field=['first_name']),
+        Reduce([], append_and_return),
+    ]).run()
+    result = controller.await_termination()
+
+    assert len(result) == 2
```

Comment:
I know the cardinality is the point, but still better to make sure we got the right two elements.